### DISCUSSION TITLE
FileNotFoundError when trying to run the exe file generated by pyinstaller

### DISCUSSION THREAD

**Discussion by pedrober31** [pyinstaller/pyinstaller#6682] [Help] [answered]
I have an interface built with QtDesigner and I want to generate an executable from it, I run the code "pyinstaller --windowed geo.py" to generate the exe file, I go to the "dist" folder and then copy the "geo" folder inside from the same to the desktop of my computer, inside the "geo" folder I click on geo.exe and I get "FileNotFoundError", even though all the path names are correct. Can you help me ?

**Comment by rokm** [comment, marked as answer]
Did you ensure that the .ui file was collected? And are you using you using proper [absolute paths](https://pyinstaller.readthedocs.io/en/stable/runtime-information.html) when trying to load the .ui file?

**Reply by pedrober31** [reply]
In the .py code there is the following line: asb = uic.loadUi("geo.ui"), both the .py and .ui code are in the same folder, so that in the development environment I can run an interface smoothly.

**Reply by rokm** [reply]
Is there a `geo.ui` next to the generated exe, though? PyInstaller won't collect it for you automatically, so you'll need to use [--add-data](https://pyinstaller.readthedocs.io/en/stable/usage.html#cmdoption-add-data) to instruct it to do so (or copy the file over manually).

But even so, your code assumes that the geo.ui (and geo.exe) are in the current working directory, which may not be the case. So you need to use absolute paths (anchored to `sys._MEIPASS` or `__file__`) to ensure that the file is always found (you can create a special codepath for frozen application, based on value of `sys.frozen`, like shown in the docs that I linked earlier).

**Reply by pedrober31** [reply]
Worked, thank you!
